components/prism-emacs.js

```diff
@@ -0,0 +1,179 @@
+Prism.languages.emacs = (function () {
+	// Patterns in regular expressions
+
+	// Symbol name. See https://www.gnu.org/software/emacs/manual/html_node/elisp/Symbol-Type.html
+	// & and : are excluded as they are usually used for special purposes
+	const symbol = '[-+*/_~!@$%^=<>{}\\w]+';
+	// symbol starting with & used in function arguments
+	const marker = '&' + symbol;
+	// Open parenthesis for look-behind
+	const par = '(\\()';
+	const endpar = '(?=\\))';
+	// End the pattern with look-ahead space
+	const space = '(?=\\s)';
+
+	// Functions to construct regular expressions
+	// simple form
+	// e.g. (interactive ... or (interactive)
+	const simple_form = name => new RegExp(`(\\()${name}(?=[\\s\\)])`);
+	// booleans and numbers
+	const primitive = pattern => new RegExp(`([\\s\\(\\[])${pattern}(?=[\\s\\)])`);
+
+	var language = {
+		// Three or four semicolons are considered a heading.
+		// See https://www.gnu.org/software/emacs/manual/html_node/elisp/Comment-Tips.html
+		'heading': {
+			pattern: /;;;.*/,
+			alias: ['comment', 'title']
+		},
+		'comment': /;.*/,
+		'string': {
+			pattern: /"(?:[^"\\]*|\\.)*"/,
```

Comment:
I think multiline feeds are supported, I say that because this is a valid function definition

```elisp
(defun multiply-by-seven (number)
       "Multiply NUMBER by seven,
second line"
       (* 7 number))
```